why-poetry.qmd

```diff
@@ -0,0 +1,97 @@
+---
+title: "Why Poetry"
+description: "Our reasons for using Poetry for managing Python package dependencies when building Django and Python projects."
+author: "Luke W. Johnston"
+date: last-modified
+categories:
+- package dependencies
+- virtual environments
+- python
+---
+
+## Context and Problem Statement
+
+Managing the packages we depend on (e.g. Django) in Python isn't "simply" installing it on your computer. The issue with installing Python packages on your computer is dependency conflicts that occur, where the "dependency trees" of one package require a different dependency tree than another one. So installing a package that requires a specific version of one package, while another package you already have installed requires another version can cause system problems.
+
+The solution to this problem is by using "virtual environments" in Python. These are environments that are separate from the rest of the system and can install any packages without impacting other environments.
+
+The problem is that there multiple ways of creating these virtual environments and managing package dependencies.
+
+## Decision Drivers
+
+- Package dependency management is a critical task, so we need a tool for it.
+- Ensuring we have some way of building our Python projects in a way that is less likely to be a "it works on my computer" issue.
+
+## Considered Options
+
+There are (unfortunately) multiple tools to manage package dependencies in Python, for example, listed [here](https://packaging.python.org/en/latest/tutorials/managing-dependencies/#other-tools-for-application-dependency-management) and [here](https://python.libhunt.com/poetry-alternatives). There are also comparison sites like [this](https://ritza.co/articles/gen-articles/pipenv-vs-virtualenv-vs-poetry-vs-pyenv-vs-pip/), [this](https://www.warp.dev/blog/prose-about-poetry), [this](https://dev.to/adamghill/python-package-manager-comparison-1g98), and [this](https://dev.to/frostming/a-review-pipenv-vs-poetry-vs-pdm-39b4) that go into more detail about many of these tools, as well as some descriptive statistics of the popularities of each [here](https://python.libhunt.com/poetry-alternatives).
```

Comment:
```suggestion
There are (unfortunately) multiple tools to manage package dependencies in Python, for example, listed at [Python Packaging User Guide](https://packaging.python.org/en/latest/tutorials/managing-dependencies/#other-tools-for-application-dependency-management) and at [Awesome Python](https://python.libhunt.com/poetry-alternatives). There are also comparison sites like [Ritza Articles](https://ritza.co/articles/gen-articles/pipenv-vs-virtualenv-vs-poetry-vs-pyenv-vs-pip/), [warp](https://www.warp.dev/blog/prose-about-poetry), and [this](https://dev.to/adamghill/python-package-manager-comparison-1g98), and [this](https://dev.to/frostming/a-review-pipenv-vs-poetry-vs-pdm-39b4) at [dev.to](https://dev.to/) that go into more detail about many of these tools, as well as some descriptive statistics of the popularities of each at [Awesome Python](https://python.libhunt.com/poetry-alternatives).
```
I prefer to use something a bit more descriptive for hyperlinks than "here" and "this". What do you think? :)
And it seems like you have added this link twice https://python.libhunt.com/poetry-alternatives. Mistake or on purpose?